src/azul/indexer/document.py

```diff
@@ -990,9 +990,11 @@ def from_index(self, value: str) -> Optional[str]:
 
 class Nested(PassThrough[JSON]):
     properties: Mapping[str, FieldType]
+    agg_property: str
 
-    def __init__(self, **properties):
+    def __init__(self, agg_property: str, /, **properties):
```

Comment:
We should assert `agg_property in properties` but I want to try a different tack: let's just assume that the first key in `properties` is the agg property.